pkg/operators/kubenetworkmanager/kubenetworkmanager.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubenetworkmanager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/kubemanager"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	ParamContainerName = "containername"
+	ParamSelector      = "selector"
+	ParamAllNamespaces = "all-namespaces"
+	ParamPodName       = "podname"
+	ParamNamespace     = "namespace"
+)
+
+type SetIPToK8sResolverInterface interface {
+	SetIPToK8sResolver(resolver func(ip string) string)
+}
+
+// TODO: Where should this be defined?
+type RemoteKind string
+
+const (
+	RemoteKindPod     RemoteKind = "pod"
+	RemoteKindService RemoteKind = "svc"
+	RemoteKindOther   RemoteKind = "other"
+)
+
+// TODO: More granualar and "shareable" interfaces
+type AdditionalPodNetworkInformation interface {
+	// Local/Host
+	SetPodOwner(string)
+	SetPodHostIP(string)
+	SetPodIP(string)
+	SetPodLabels(map[string]string)
+
+	// Remote
+	GetRemoteIP() string
+	SetRemoteNamespace(string)
+	SetRemoteKind(string)
+	SetRemotePodLabels(map[string]string)
+}
+
+type KubeNetworkManager struct {
+	clientset *kubernetes.Clientset
+}
+
+func (k *KubeNetworkManager) Name() string {
+	return "KubeNetworkManager"
+}
+
+func (k *KubeNetworkManager) Description() string {
+	return "KubeNetworkManager handles more descriptive informations about remote ips"
+}
+
+func (k *KubeNetworkManager) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (k *KubeNetworkManager) ParamDescs() params.ParamDescs {
+	// TODO: Write patch so empty paramsDescs is okay
+	return params.ParamDescs{
+		{
+			Key:         "asdfsadf",
+			Description: "panics without at least one param",
+		},
+	}
+}
+
+func (k *KubeNetworkManager) Dependencies() []string {
+	return []string{"KubeManager"}
+}
+
+func (k *KubeNetworkManager) CanOperateOn(gadget gadgets.GadgetDesc) bool {
+	km := kubemanager.KubeManager{}
+	if !km.CanOperateOn(gadget) {
+		return false
+	}
```

Comment:
Currently we need this check afaik (crashes without the check), but it would be a nice feature maybe that it could be inferred by the `Dependencies()`

Reply:
The `Dependencies()` only define the order in which operators are called right now, not whether they are called. Like @burak-ok said, we should probably add that as an additional check (not in this PR, though).

I think the crash would be coming by not having a check for the `operators.ContainerInfoGetters` in the `CanOperatorOn()` and just casting to it in `enrich()`. IMHO, it would be better to have that check locally as well and remove the import of the kubemanager.

Reply:
The crash is originating from sorting the operators. `KubeNetworkManager` wants `KubeManager` but it can not be found in the graph.
-> Panic is intended here